File: Week-1/Day3/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = nums[0];
        int prev = nums[0];
        for(int i=1; i<nums.size(); i++) {
            int curr_max = max(nums[i], nums[i] + prev);
            prev = curr_max;
            ans = max(ans, curr_max);
        }
        return ans;
    }
};

File: Week-1/Day3/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxAns, curMax = nums[0], 0
        for num in nums:
            curMax = max(curMax + num, num)
            maxAns = max(maxAns, curMax)
        
        return maxAns
            